src/NUnitFramework/tests/Internal/UnexpectedExceptionTests.cs

```diff
@@ -100,24 +97,125 @@ public void FailRecordsNestedInnerExceptionAsPartOfAggregateException()
         [Test]
         public void BadStackTraceIsHandled()
         {
-            ITestResult result = TestBuilder.RunTestCase(
-                typeof(UnexpectedExceptionFixture),
-                "ThrowsWithBadStackTrace");
+            ITestResult result = RunDataTestCase(nameof(UnexpectedExceptionFixture.ThrowsWithBadStackTrace));
 
             Assert.AreEqual(ResultState.Error, result.ResultState);
             Assert.AreEqual("NUnit.TestData.UnexpectedExceptionFixture.ExceptionWithBadStackTrace : thrown by me", result.Message);
-            Assert.AreEqual("No stack trace available", result.StackTrace);
+            Assert.AreEqual("InvalidOperationException was thrown by the Exception.StackTrace property.", result.StackTrace);
         }
 
         [Test]
         public void CustomExceptionIsHandled()
         {
-            ITestResult result = TestBuilder.RunTestCase(
-                typeof(UnexpectedExceptionFixture),
-                "ThrowsCustomException");
+            ITestResult result = RunDataTestCase(nameof(UnexpectedExceptionFixture.ThrowsCustomException));
 
             Assert.AreEqual(ResultState.Error, result.ResultState);
             Assert.AreEqual("NUnit.TestData.UnexpectedExceptionFixture.CustomException : message", result.Message);
         }
+
+        [Test]
+        public void AssertThatCanHandleRecursivelyThrowingExceptionAsActual()
+        {
+            var result = (ITestResult)null;
+
+            var ex = RecordPossiblyDangerousException(() =>
+                result = RunDataTestCase(nameof(UnexpectedExceptionFixture.AssertThatWithRecursivelyThrowingExceptionAsActual)));
+
+            Assert.That(ex is null); // Careful not to pass ex to Assert.That and crash the test run rather than failing
+
+            Assert.That(result, Has.Property("Message").StartWith(
+                "  Expected: null" + Environment.NewLine
+                + "  But was:  <! RecursivelyThrowingException was thrown by RecursivelyThrowingException.ToString() !>"));
```

Comment:
@nunit/framework-team Do you like the way this looks? I added `!` and an extra space on each side to indicate that this is not a normal situation.

Reply:
I think it is fine (and I cannot come up with anything better)